diff -rupN xv6-orig/param.h xv6-1.1/param.h
--- xv6-orig/param.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/param.h	2018-04-04 15:56:26.756561439 -0300
@@ -10,4 +10,5 @@
 #define MAXOPBLOCKS  10  // max # of blocks any FS op writes
 #define LOGSIZE      (MAXOPBLOCKS*3)  // max data sectors in on-disk log
 #define NBUF         (MAXOPBLOCKS*3)  // size of disk block cache
+#define QUANTUM		  10  // maximum number of ticks for process 
 
diff -rupN xv6-orig/proc.h xv6-1.1/proc.h
--- xv6-orig/proc.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/proc.h	2018-03-28 15:17:27.937613000 -0300
@@ -66,6 +66,7 @@ struct proc {
   struct file *ofile[NOFILE];  // Open files
   struct inode *cwd;           // Current directory
   char name[16];               // Process name (debugging)
+  uint pticks;                 // Local ticks to the process
 };
 
 // Process memory is laid out contiguously, low addresses first:
diff -rupN xv6-orig/syscall.c xv6-1.1/syscall.c
--- xv6-orig/syscall.c	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/syscall.c	2018-04-04 15:47:45.876557069 -0300
@@ -98,6 +98,7 @@ extern int sys_unlink(void);
 extern int sys_wait(void);
 extern int sys_write(void);
 extern int sys_uptime(void);
+extern int sys_procstat(void);
 
 static int (*syscalls[])(void) = {
 [SYS_fork]    sys_fork,
@@ -121,6 +122,7 @@ static int (*syscalls[])(void) = {
 [SYS_link]    sys_link,
 [SYS_mkdir]   sys_mkdir,
 [SYS_close]   sys_close,
+[SYS_procstat]   sys_procstat,
 };
 
 void
diff -rupN xv6-orig/syscall.h xv6-1.1/syscall.h
--- xv6-orig/syscall.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/syscall.h	2018-04-04 15:45:28.292555915 -0300
@@ -20,3 +20,4 @@
 #define SYS_link   19
 #define SYS_mkdir  20
 #define SYS_close  21
+#define SYS_procstat  22
diff -rupN xv6-orig/sysproc.c xv6-1.1/sysproc.c
--- xv6-orig/sysproc.c	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/sysproc.c	2018-04-05 12:14:32.188319710 -0300
@@ -89,3 +89,12 @@ sys_uptime(void)
   release(&tickslock);
   return xticks;
 }
+
+// Proyecto1
+// List the existing system's processes and their status
+int
+sys_procstat(void)
+{
+  procdump();
+  return 0;
+}
\ No hay ningún carácter de nueva línea al final del archivo
diff -rupN xv6-orig/trap.c xv6-1.1/trap.c
--- xv6-orig/trap.c	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/trap.c	2018-04-05 11:28:00.164391315 -0300
@@ -103,7 +103,13 @@ trap(struct trapframe *tf)
   // Force process to give up CPU on clock tick.
   // If interrupts were on while locks held, would need to check nlock.
   if(proc && proc->state == RUNNING && tf->trapno == T_IRQ0+IRQ_TIMER)
-    yield();
+    // Proyecto0
+    // Force process to give up CPU on the clock ticks given by quantum.
+    if(proc->pticks++ == QUANTUM-1){
+      //cprintf("Proceso: %d terminado en %d ticks\n", proc->pid, proc->pticks );
+      proc->pticks = 0;
+      yield();
+    }
 
   // Check if the process has been killed since we yielded
   if(proc && proc->killed && (tf->cs&3) == DPL_USER)
diff -rupN xv6-orig/user/forktest.c xv6-1.1/user/forktest.c
--- xv6-orig/user/forktest.c	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/user/forktest.c	2018-04-04 15:41:24.080553866 -0300
@@ -21,6 +21,8 @@ forktest(void)
   printf(1, "fork test\n");
 
   for(n=0; n<N; n++){
+    if(n==5)
+      procstat();
     pid = fork();
     if(pid < 0)
       break;
diff -rupN xv6-orig/user/param.h xv6-1.1/user/param.h
--- xv6-orig/user/param.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/user/param.h	2018-04-04 15:56:26.756561439 -0300
@@ -10,4 +10,5 @@
 #define MAXOPBLOCKS  10  // max # of blocks any FS op writes
 #define LOGSIZE      (MAXOPBLOCKS*3)  // max data sectors in on-disk log
 #define NBUF         (MAXOPBLOCKS*3)  // size of disk block cache
+#define QUANTUM		  10  // maximum number of ticks for process 
 
diff -rupN xv6-orig/user/syscall.h xv6-1.1/user/syscall.h
--- xv6-orig/user/syscall.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/user/syscall.h	2018-04-04 15:45:28.292555915 -0300
@@ -20,3 +20,4 @@
 #define SYS_link   19
 #define SYS_mkdir  20
 #define SYS_close  21
+#define SYS_procstat  22
diff -rupN xv6-orig/user/user.h xv6-1.1/user/user.h
--- xv6-orig/user/user.h	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/user/user.h	2018-04-04 15:46:48.660556589 -0300
@@ -23,6 +23,7 @@ int getpid(void);
 char* sbrk(int);
 int sleep(int);
 int uptime(void);
+int procstat(void);
 
 // ulib.c
 int stat(char*, struct stat*);
diff -rupN xv6-orig/user/usys.S xv6-1.1/user/usys.S
--- xv6-orig/user/usys.S	2015-04-08 15:02:06.000000000 -0300
+++ xv6-1.1/user/usys.S	2018-04-04 15:45:51.728556112 -0300
@@ -29,3 +29,4 @@ SYSCALL(getpid)
 SYSCALL(sbrk)
 SYSCALL(sleep)
 SYSCALL(uptime)
+SYSCALL(procstat)
